express-notes-api/index.js

```diff
@@ -0,0 +1,95 @@
+const data = require('./data.json');
+const express = require('express');
+const fs = require('fs');
+
+const app = express();
+
+app.get('/api/notes', (req, res) => {
+  const notesArray = [];
+  for (const entries in data.notes) {
+    notesArray.push(data.notes[entries]);
+  }
+  res.status(200).send(notesArray);
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const id = Number(req.params.id);
+  if (id < 0 || !Number.isInteger(id)) {
+    res.status(400).json({ error: 'id must be a positive integer' });
+  }
+  if (id >= 0 && data.notes[id] !== undefined) {
+    res.status(200).json(data.notes[id]);
+  } else {
+    res.status(404).send({ error: `cannot find note with id ${id}` });
+  }
+});
+
+app.use(express.json());
+
+app.post('/api/notes', (req, res) => {
+  const newEntry = req.body;
+  if (!newEntry.content) {
+    res.status(400).json({ error: 'content is a required field' });
+  } else {
+    newEntry.id = data.nextId;
+    data.notes[data.nextId] = newEntry;
+    data.nextId++;
+    const updatedData = JSON.stringify(data, null, 2);
+    fs.writeFile('./data.json', updatedData, 'utf8', err => {
+      if (err) {
+        console.error({ error: 'An unexpected error occurred.' });
+        res.sendStatus(500);
```

Comment:
All of your 500 scenarios need to

- log the `err` to the console with `console.error()`
- send back a JSON 500 response to the client `{ error: 'An unexpected error occurred' }`.